pkg/upgrade/control_plane.go

```diff
@@ -33,6 +33,17 @@ type ControlPlaneUpgrader struct {
 	oldNodeToEtcdMember map[string]string
 }
 
+func NewTestableControlPlaneUpgrader(log logr.Logger, cmn ConfigMapNamespacer, pn PodNamespacer, nc NodeClient) *ControlPlaneUpgrader {
```

Comment:
I'd assert that newBase is too smart and the logic should be kept somewhere else. probably some client factory would be sufficient.